frontend/src/containers/Loading.js

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import LoadingComponent from '../components/Loading'
+
+const nextPosition = (position, max, direction) => {
+  if (direction < 0 && position <= 0) return 1
+  if (direction > 0 && position >= max - 1) return max - 2
+  return position + direction
+}
+
+const nextDirection = (position, max, direction) => {
+  if (position <= 0 || position >= max - 1) return direction * -1
+  return direction
+}
+
+class Loading extends Component {
+  state = {
+    position: 0,
+    direction: 1,
+  }
+
+  componentDidMount() {
+    this.intervalId = setInterval(() => {
+      this.setState(({ position, direction }) => ({
+        position: nextPosition(position, this.props.size, direction),
+        direction: nextDirection(position, this.props.size, direction),
+      }))
+    }, 50)
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.intervalId)
+  }
+
+  render() {
+    const { size = 60 } = this.props
+    const characters = new Array(size).fill('\u00A0')
+
+    return (<div class="loading-bar">[<LoadingComponent position={this.state.position} characters={characters} />]</div>)
+  }
+}
+
+export default Loading;
```

Comment:
I'm think this whole component is way too complex to solve a simple loading problem... seriously :S a pure css solution could be used: https://codepen.io/thetallweeks/pen/yybGra

Reply:
I think it will be harder to do animation with CSS, because the animation should be:

[=           ]
[ =          ]
[  =         ]
[   =        ]
[    =       ]
[     =      ]
[      =     ]
[       =    ]
[        =   ]
[         =  ]
[          = ]
[           =]
[          = ]
[         =  ]
[        =   ]
[       =    ]
[      =     ]
[     =      ]
[    =       ]
[   =        ]
[  =         ]
[ =          ]
[=           ]

Reply:
Hm, I'll try create a @keyframes for it :stuck_out_tongue:

Reply:
Done: https://codepen.io/rafaelverger/pen/rvwxeQ

Reply:
With some easy in and out will be awesome also :3

Reply:
@rafaelverger I made a small change: https://codepen.io/RodolfoSilva/pen/LmLNez 
![image](https://user-images.githubusercontent.com/1703020/39554357-b163676c-4e48-11e8-9201-489617ed9be0.png)

Reply:
@rafaelverger thanks 😛

Reply:
@RodolfoSilva nice changes.. the borders i've set wasn't necessary to be in the final code, was meant just to see the character running from beginning to the end 👍